File: SmartDelivery/app/src/main/java/com/example/smartdelivery/ui/main/view/activity/SignupActivity.java
package com.example.smartdelivery.ui.main.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.toolbox.Volley;
import com.example.smartdelivery.R;
import com.example.smartdelivery.ui.main.request.RegisterRequest;

import org.json.JSONException;
import org.json.JSONObject;

import androidx.appcompat.app.AppCompatActivity;

public class SignupActivity extends AppCompatActivity {
//회원가입
    private EditText userid,userpw,username,userphone,useraddress,useremail;
    private Button sign_btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);

        userid=findViewById(R.id.userid);
        userpw=findViewById(R.id.userpw);
        username=findViewById(R.id.username);
        userphone=findViewById(R.id.userphone);
        useraddress=findViewById(R.id.useraddress);
        useremail=findViewById(R.id.useremail);

        sign_btn=findViewById(R.id.sign_btn);

        sign_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String USERID=userid.getText().toString();
                String USERPW=userpw.getText().toString();
                String USERNAME=username.getText().toString();
                String USERPHONE=userphone.getText().toString();
                String USERADDRESS=useraddress.getText().toString();
                String USEREMAIL=useremail.getText().toString();

                Response.Listener<String> responseListener = new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {

                        try {
                            JSONObject jsonObject = new JSONObject( response );
                            boolean success = jsonObject.getBoolean( "success" );

                            //회원가입 성공시
                            if(success) {

                                Toast.makeText( getApplicationContext(), "성공", Toast.LENGTH_SHORT ).show();
//                                Intent intent = new Intent( SignupActivity.this, testActivity.class );
//                                startActivity( intent );


                                //회원가입 실패시
                            } else {
                                Toast.makeText( getApplicationContext(), "실패", Toast.LENGTH_SHORT ).show();
                                return;
                            }

                        } catch (JSONException e) {
                            e.printStackTrace();
                        }

                    }
                };
                //서버로 Volley를 이용해서 요청
                RegisterRequest registerRequest = new RegisterRequest(USERID,USERPW,USERNAME,USERPHONE,USERADDRESS,USEREMAIL, responseListener);
                RequestQueue queue = Volley.newRequestQueue( SignupActivity.this );
                queue.add( registerRequest );

            }
        });
    }

}

File: SmartDelivery/app/src/main/java/com/example/smartdelivery/ui/main/view/activity/SignupActivity.kt
package com.example.smartdelivery.ui.main.view.activity

import android.content.Intent

import android.util.Log
import com.example.smartdelivery.R
import com.example.smartdelivery.data.model.request.SignupRequest
import com.example.smartdelivery.data.model.response.SignupResponse
import com.example.smartdelivery.data.shared.MySharedPreferences
import com.example.smartdelivery.databinding.ActivitySignupBinding
import com.example.smartdelivery.ui.base.BaseActivity
import com.example.smartdelivery.ui.main.view.dialog.ProgressDialog
import com.example.smartdelivery.ui.main.viewmodel.SignupViewModel
import kotlinx.android.synthetic.main.activity_login.*
import org.koin.androidx.viewmodel.ext.android.viewModel

class SignupActivity : BaseActivity<ActivitySignupBinding>(R.layout.activity_signup){
//회원가입
    private var signupResponse=SignupResponse()
    private var signupRequest=SignupRequest()
    private val signupViewModel:SignupViewModel by viewModel()

    private val dialog by lazy {
        ProgressDialog(this)
    }
    private val TAG = "SignupActivity"

// 회원가입

    override fun init() {
        super.init()
        binding.signBtn.setOnClickListener {
            signupRequest.id = binding.eId.text.toString()
            signupRequest.password = binding.ePw.text.toString()
            signupRequest.name = binding.eName.text.toString()
            signupRequest.phone = binding.ePhone.text.toString()
            signupRequest.address = binding.eAddress.text.toString()
            signupRequest.email = binding.eEmail.text.toString()
            initViewModel()

            startActivity(Intent(this, SignupSuccessActivity::class.java))



        }

    }

    //회원가입 서버API 통신
    private fun initViewModel(){
        signupViewModel.SignupApiCall(signupRequest).observe(this@SignupActivity,
        androidx.lifecycle.Observer { resource->
            when(resource.status) {
                com.example.smartdelivery.util.Resource.Status.SUCCESS -> {
                    dialog.dismiss()
                    when (resource.data!!.code()) {
                        200 -> {
                            Log.d(TAG, "initViewModel: ${resource.data.body()!!}")
                            signupResponse = resource.data.body()!!
                        }
                        else -> {
                            toast(this, "에러요")
                        }
                    }
                }
            }
        })


}

    private  fun signupApi(){
        if (signupResponse.isSuccess){
            MySharedPreferences.setUserId(this@SignupActivity,binding.eId.text.toString())
            MySharedPreferences.setUserPass(this@SignupActivity,binding.ePw.text.toString())
            MySharedPreferences.setUserName(this@SignupActivity,binding.eName.text.toString())
            MySharedPreferences.setUserAddress(this@SignupActivity,binding.eAddress.text.toString())
            MySharedPreferences.setUserPhone(this@SignupActivity,binding.ePhone.text.toString())
            MySharedPreferences.setUserEmail(this@SignupActivity,binding.eEmail.text.toString())
            startActivity(Intent(this@SignupActivity, MenuActivity::class.java))
            finish()


        }

    }    }